model/common/src/icon4py/model/common/metrics/compute_diffusion_metrics.py

```diff
@@ -115,34 +81,29 @@ def _compute_k_start_end(
     c_owner_mask: data_alloc.NDArray,
     thslp_zdiffu: float,
     thhgtd_zdiffu: float,
-    cell_nudging: int,
-    n_cells: int,
     nlev: int,
     array_ns: ModuleType = np,
-) -> tuple[data_alloc.NDArray, data_alloc.NDArray]:
-    k_start = array_ns.zeros(n_cells, dtype=int)
-    k_end = array_ns.zeros(n_cells, dtype=int)
-    for jc in range(cell_nudging, n_cells):
-        if (
-            maxslp_avg[jc, nlev - 1] >= thslp_zdiffu or maxhgtd_avg[jc, nlev - 1] >= thhgtd_zdiffu
-        ) and c_owner_mask[jc]:
-            for jk in reversed(range(nlev)):
-                if z_mc[jc, jk] >= max_nbhgt[jc]:
-                    k_end[jc] = jk + 1
-                    break
-
-            for jk in range(nlev):
-                if maxslp_avg[jc, jk] >= thslp_zdiffu or maxhgtd_avg[jc, jk] >= thhgtd_zdiffu:
-                    k_start[jc] = jk
-                    break
+) -> tuple[data_alloc.NDArray, data_alloc.NDArray, data_alloc.NDArray]:
+    condition1 = array_ns.logical_or(maxslp_avg >= thslp_zdiffu, maxhgtd_avg >= thhgtd_zdiffu)
+    cell_mask = array_ns.tile(
+        array_ns.where(condition1[:, nlev - 1], c_owner_mask, False), (nlev, 1)
+    ).T
+    threshold = array_ns.tile(max_nbhgt, (nlev, 1)).T
+    owned_cell_above_threshold = array_ns.logical_and(cell_mask, z_mc >= threshold)
+    last_true_indices = nlev - 1 - array_ns.argmax(owned_cell_above_threshold[:, ::-1], axis=1)
+    # is a zero else value correct?
+    kend = array_ns.where(
+        array_ns.any(owned_cell_above_threshold, axis=1), last_true_indices + 1, 0
+    )
 
-            if all((k_start[jc], k_end[jc])) and k_start[jc] > k_end[jc]:
-                k_start[jc] = nlev - 1
+    kstart = np.argmax(condition1, axis=1)
+    # reset the values where start > end to be an empty range(start, end)
+    kstart = array_ns.where(kstart > kend, nlev, kstart)
+    cell_index_cell_mask = array_ns.where(kend > kstart, True, False)
 
-    return k_start, k_end
+    return kstart, kend, cell_index_cell_mask
```

Comment:
```suggestion
    return kstart, kend, cell_index_mask
```
Just as a suggestion